components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/util/APIErrorResponseHandler.java

```diff
@@ -53,13 +59,18 @@ public static void handleErrorResponse(AuthenticationContext authenticationConte
         if (log.isDebugEnabled() && e != null) {
             log.debug("Authentication Error ", e);
         }
-        StringBuilder value = new StringBuilder(16);
-        value.append("realm user=\"");
-        if (authenticationContext != null && authenticationContext.getUser() != null) {
-            value.append(authenticationContext.getUser().getUserName());
+        if (error == HttpServletResponse.SC_UNAUTHORIZED) {
+            response.setHeader(AUTH_HEADER_NAME, getRealmInfo());
+        }
+        else if (!removeRealmUser){
```

Comment:
formatting issue. `else if` should move to line 64

Reply:
Addressed with https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/233